src/analysis/individualStudy/management/DataManagementAccordionItem.tsx

```diff
@@ -37,20 +44,32 @@ export function DataManagementAccordionItem({ studyId, refresh }: { studyId: str
 
   const handleCreateSnapshot = async () => {
     setLoading(true);
-    await storageEngine.createSnapshot(studyId, false);
-    refreshSnapshots();
-    setLoading(false);
-    await refresh();
+    const createdSnapshot: FirebaseActionResponse = await storageEngine.createSnapshot(studyId, false);
+    if (createdSnapshot.status === 'SUCCESS') {
+      refreshSnapshots();
+      setLoading(false);
+      await refresh();
+    } else {
+      setLoading(false);
+      setError(createdSnapshot.error);
+      setModalErrorOpened(true);
+    }
   };
 
   const handleArchiveData = async () => {
     setLoading(true);
     setDeleteValue('');
     setModalArchiveOpened(false);
-    await storageEngine.createSnapshot(studyId, true);
-    refreshSnapshots();
-    setLoading(false);
-    await refresh();
+    const createdSnapshot: FirebaseActionResponse = await storageEngine.createSnapshot(studyId, true);
+    if (createdSnapshot.status === 'SUCCESS') {
+      refreshSnapshots();
+      setLoading(false);
+      await refresh();
+    } else {
+      setLoading(false);
+      setError(createdSnapshot.error);
+      setModalErrorOpened(true);
+    }
```

Comment:
Is it worth refactoring this repeated block into another function in this file?

Reply:
Thought about this too. Stuff is getting a little burdensome. I'll see how it feels when I try to refactor.